lib/utils/aws/signing.go

```diff
@@ -120,9 +123,32 @@ func (sc *SigningCtx) Check(clock clockwork.Clock) error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
+
+	sc.SessionName = MaybeHashRoleSessionName(sc.SessionName)
 	return nil
 }
 
+// MakeSessionMetadata returns common AWS session metadata.
+func (sc *SigningCtx) MakeSessionMetadata() events.AWSSessionMetadata {
+	if sc == nil {
+		return events.AWSSessionMetadata{}
+	}
+	return MakeSessionMetadata(sc.SessionName, sc.AWSExternalID, sc.SessionTags)
+}
+
+// MakeSessionMetadata returns common AWS session metadata.
+func MakeSessionMetadata(roleSessionName, externalID string, sessionTags map[string]string) events.AWSSessionMetadata {
+	e := events.AWSSessionMetadata{
+		RoleSessionName: roleSessionName,
+		SessionTags:     sessionTags,
+	}
+	if externalID != "" {
+		sum := sha1.Sum([]byte(externalID))
+		e.ExternalIdSha1 = hex.EncodeToString(sum[:])
```

Comment:
Why externalID needs to be truncated/ converted to sha1 ?

Reply:
Here is my take. AWS does not treat external ID as a secret but in general, it's only known to the assumer and whoever has permission to the assumed role. It should not be public knowledge to like an auditor. 

For auditing purpose, I mainly want to capture the fact that an external ID is used. I could also use a bool.

What do you think? If security is not a concern, we could record the original string too.